### PR TITLE
Enhanced JSON 'navigation' and fixed #617

### LINKED ISSUE #617: AWS NLB - Lack of Access log (false positive)
Scout is giving a false positive on "Lack of Access Log" for NLB configured without a TLS listener.

It's not possible to configure Access log with the above configuration

As per AWS documentation: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-access-logs.html

"Access logs are created only if the load balancer has a TLS listener and they contain information only about TLS requests."

### PR DESCRIPTION
**Pull request by alessandrogonzalez**
# Description

Fixes https://github.com/nccgroup/ScoutSuite/issues/617

Map a new synthetic field in elbv2 listeners called `listener_protocols`, which is a list of protocols used in the load balancer.

## Type of change

Select the relevant option(s):

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Aboisier** [comment]
# [Codecov](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=h1) Report
> Merging [#656](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=desc) into [develop](https://codecov.io/gh/nccgroup/ScoutSuite/commit/4f57c99e35972f5da568f57702af537f5ffbc7f6&el=desc) will **decrease** coverage by `0.16%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #656      +/-   ##
===========================================
- Coverage    65.21%   65.04%   -0.17%     
===========================================
  Files           22       22              
  Lines         1515     1522       +7     
===========================================
+ Hits           988      990       +2     
- Misses         527      532       +5     
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=footer). Last update [4f57c99...3520da4](https://codecov.io/gh/nccgroup/ScoutSuite/pull/656?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: ScoutSuite/providers/aws/rules/findings/elbv2-no-access-logs.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "display_path": "elbv2.regions.id.vpcs.id.lbs.id",
     "conditions": [ "and",
         [ "Key", "equal", "access_logs.s3.enabled" ],
-        [ "Value", "equal", "false"]
+        [ "Value", "equal", "false"],
+        ["elbv2.regions.id.vpcs.id.lbs._GET_VALUE_AT_(elbv2.regions.id.vpcs.id.lbs.id).listeners.__values__.__mapParam__Protocol", "containAtLeastOneOf", ["TLS", "HTTP", "HTTPS"]]
```
Comment (j4v) [inline ScoutSuite/providers/aws/rules/findings/elbv2-no-access-logs.json]: Shouldn't it just be `["TLS", "HTTPS"]`?

**Comment by j4v** [comment]
@alessandrogonzalez wouldn't it make more sense to define the `__values__` & `__mapParam__ParamName` in `ScoutSuite/core/conditions.py`, akin to _GET_VALUE_AT_.

Looking at the big picture, I get the feeling your solution is very specific to this issue and the root problem is the poor / non-existant parsing in `ScoutSuite/providers/aws/resources/elbv2/load_balancers.py` and `ScoutSuite/providers/aws/resources/elbv2/listeners.py`. 

If the `attributes` field was remapped as a dict, and `listeners` as a list, it would be much easier to write rules for `load_balancers`. Granted this would require refactoring a few rules, but I think it would be simpler then adding a condition to `browser.py`. 

I might be wrong, happy to get your thoughts.

**Comment by alessandrogonzalez** [comment]
I agree that this looks like a solution for a specific problem that can be solved by better parsing the information. I will rethink this and update this PR.

File: ScoutSuite/providers/aws/rules/findings/elbv2-no-access-logs.json (hunk 1 above)
Comment (alessandrogonzalez) [reply ScoutSuite/providers/aws/rules/findings/elbv2-no-access-logs.json]: Issue 617 is related to network load balancers. In the documentation of access logging for application load balancers (https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html) there is no mention that only the HTTPS listeners are getting logged, so I assumed that both of them (HTTP and HTTPS) will be logged, so the issue should be applicable to them.

**Comment by j4v** [comment]
Closing, this has now been ghetto-merged in `develop`.

**Pull request closed without merging**